sdk/ai/azure-ai-projects/samples/agents/async_samples/sample_agents_code_interpreter_attachment_enterprise_search_async.py

```diff
@@ -0,0 +1,97 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""
+FILE: sample_agents_code_interpreter_attachment.py
+
+DESCRIPTION:
+    This sample demonstrates how to use agent operations with code interpreter from
+    the Azure Agents service using a synchronous client.
+
+USAGE:
+    python sample_agents_code_interpreter_attachment.py
+
+    Before running the sample:
+
+    pip install azure.ai.projects azure-identity
+
+    Set this environment variables with your own values:
+    PROJECT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+"""
+import asyncio
+import os
+from azure.ai.ml._ml_client import MLClient
+from azure.ai.ml.constants import AssetTypes
+from azure.ai.ml.entities import Data
+
+from azure.ai.projects.aio import AIProjectClient
+from azure.ai.projects.models import (
+    CodeInterpreterTool,
+    MessageAttachment,
+    VectorStoreDataSource,
+    VectorStoreDataSourceAssetType,
+)
+from azure.identity import DefaultAzureCredential
+
+
+# Create an Azure AI Client from a connection string, copied from your AI Studio project.
+# At the moment, it should be in the format "<HostName>;<AzureSubscriptionId>;<ResourceGroup>;<HubName>"
+# Customer needs to login to Azure subscription via Azure CLI and set the environment variables
+
+
+async def main():
+    credential = DefaultAzureCredential()
+    project_client = AIProjectClient.from_connection_string(
+        credential=credential, conn_str=os.environ["PROJECT_CONNECTION_STRING"]
+    )
+
+    async with project_client:
+
+        code_interpreter = CodeInterpreterTool()
+
+        # notice that CodeInterpreter must be enabled in the agent creation, otherwise the agent will not be able to see the file attachment
+        agent = await project_client.agents.create_agent(
+            model="gpt-4-1106-preview",
+            name="my-assistant",
+            instructions="You are helpful assistant",
+            tools=code_interpreter.definitions,
+        )
+        print(f"Created agent, agent ID: {agent.id}")
+
+        thread = await project_client.agents.create_thread()
+        print(f"Created thread, thread ID: {thread.id}")
+
+        ml_client = MLClient.from_config(credential)
```

Comment:
Please update to use upload_file method from projects client

Reply:
fixed.